boa_engine/src/value/operations.rs

```diff
@@ -449,14 +461,14 @@ impl JsValue {
         }
     }
 
+    /// Returns the negated value.
     #[inline]
     pub fn neg(&self, context: &mut Context) -> JsResult<Self> {
         Ok(match *self {
             Self::Symbol(_) | Self::Undefined => Self::new(f64::NAN),
-            Self::Object(_) => Self::new(match self.to_numeric_number(context) {
-                Ok(num) => -num,
-                Err(_) => f64::NAN,
-            }),
+            Self::Object(_) => {
+                Self::new(self.to_numeric_number(context).map_or(f64::NAN, |num| -num))
```

Comment:
Here I would use `f64::neg` one the second part, which should more clearly indicate intent